### PR TITLE
Fix lints

### PR DESCRIPTION
**Pull request by amab8901**
Addresses https://github.com/theseus-os/Theseus/issues/526

### CODE REVIEW

File: kernel/wait_queue/src/lib.rs [hunk 1]
```diff
@@ -236,3 +236,9 @@ impl WaitQueue {
         }
     }
 }
+
+impl Default for WaitQueue {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```
Comment (kevinaboos) [inline kernel/wait_queue/src/lib.rs]: why is this needed? If it's not, please remove.

**Review by kevinaboos — CHANGES_REQUESTED** [review]
Thanks! Looks good overall. 

Kindly undo the changes the `kernel/debug_info` and `waitqueue`, which seem accidental and unrelated to this PR.

File: kernel/wait_queue/src/lib.rs (hunk 1 above)
Comment (amab8901) [reply kernel/wait_queue/src/lib.rs]: Clippy recommended it

**Comment by amab8901** [comment]
> Thanks! Looks good overall.
> 
> Kindly undo the changes the `kernel/debug_info` and `waitqueue`, which seem accidental and unrelated to this PR.

What about these lints?
https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
https://rust-lang.github.io/rust-clippy/master/index.html#needless_doctest_main

File: kernel/wait_queue/src/lib.rs (hunk 1 above)
Comment (kevinaboos) [reply kernel/wait_queue/src/lib.rs]: is it actually used somewhere? I don't want to add or derive a bunch of unnecessary trait impls until they are actually needed.

Comment (kevinaboos) [reply kernel/wait_queue/src/lib.rs]: If this is the only place where the lint triggers, I suppose it's fine to add it. But if this is a frequent pattern, I'd rather just disable the entire lint. Code isn't free, this style lint is rather silly imo.

**Comment by kevinaboos** [comment]
Thanks for responding, see my comments on discord -- you could try changing `no_run` to `ignore` and/or changing `fn main` to `fn foo`

**Review by kevinaboos — APPROVED** [review]
